app/models/api_user.rb

```diff
@@ -8,6 +8,10 @@ class ApiUser < ApplicationRecord
 
   has_secure_token :token, length: 36
 
+  default_scope { where(revoked_at: nil) }
+  scope :with_revoked, -> { unscope(where: :revoked_at) }
```

Comment:
How about just an `active` and `revoked` scope?

scope :active, -> { where(revoked_at: nil) }
scope :revoked, -> { where.not(revoked_at: nil) }

Then can update the authenticate class method with `active.find_by(token: token)`